src/protocols/rdp/guac_rdpdr/rdpdr_printer.c

```diff
@@ -141,18 +143,25 @@ static void guac_rdpdr_device_printer_announce_handler(guac_rdpdr_device* device
     Stream_Write(output_stream, "PRN1\0\0\0\0", 8); /* DOS name */
 
     /* Printer data */
-    Stream_Write_UINT32(output_stream, 24 + GUAC_PRINTER_DRIVER_LENGTH + GUAC_PRINTER_NAME_LENGTH);
+    int settings_length = guac_utf8_strlen(device->device_name);
+    int printer_name_length = (settings_length + 1) * 2;
+    char* printer_name = malloc(printer_name_length);
```

Comment:
If you will be dynamically-allocating the printer name here, it will need to be freed. Alternatively, statically-allocating the buffer may simplify things.